reconcile/vault_replication.py

```diff
@@ -0,0 +1,176 @@
+    for policy in query_data.policy:
+        if (
+            isinstance(policy, VaultPolicyV1)
+            and policy.name == policy_name
+            and policy.instance.name == instance_name
+        ):
+            for line in policy.rules.split("\n"):
+                res = re.search(r"path \s*[\'\"](.*)[\'\"]", line)
+
+                if res is not None:
+                    policy_paths.append(res.group(1))
+
+    return policy_paths
+
+
+def get_jenkins_secret_list() -> List[str]:
+    secret_list = []
+    query_data = jenkins_configs.query(query_func=gql.get_api().query)
+
+    for p in query_data.jenkins_configs:
+        if p.config_path and p.config_path.content != "":
+            secret_paths = [
+                line
+                for line in p.config_path.content.split("\n")
+                if "secret-path" in line
+            ]
+            for line in secret_paths:
+                res = re.search(r"secret-path:\s*[\'\"](.*)[\'\"]", line).group(1)
+                secret_list.append(res)
+
+    return secret_list
+
+
+def get_vault_credentials(vault_instance: vault_instances.VaultInstanceV1):
+    vault_creds = {"server": None, "role_id": None, "secret_id": None}
+    vault = VaultClient()
+
+    if isinstance(vault_instance, VaultReplicationV1):
+        vault_instance = vault_instance.instance
+
+    role_id = {
+        "path": vault_instance.auth.role_id.path,
+        "field": vault_instance.auth.role_id.field,
+    }
+    secret_id = {
+        "path": vault_instance.auth.secret_id.path,
+        "field": vault_instance.auth.secret_id.field,
+    }
+
+    vault_creds["role_id"] = vault.read(role_id)
+    vault_creds["secret_id"] = vault.read(secret_id)
+    vault_creds["server"] = vault_instance.address
+
+    return vault_creds
+
+
+def run(dry_run: bool) -> None:
+
+    query_data = vault_instances.query(query_func=gql.get_api().query)
+
+    for instance in query_data.vault_instances:
+        if instance.replication is not None:
+            for replication in instance.replication:
+                source_creds = get_vault_credentials(instance)
+                dest_creds = get_vault_credentials(instance.replication[0])
+
+                source_vault = VaultClient(
+                    server=source_creds["server"],
+                    role_id=source_creds["role_id"],
+                    secret_id=source_creds["secret_id"],
+                )
+                dest_vault = VaultClient(
+                    server=dest_creds["server"],
+                    role_id=dest_creds["role_id"],
+                    secret_id=dest_creds["secret_id"],
+                )
+
+                provider = replication.provider
+
+                if provider == "jenkins":
+                    if replication.policy is not None:
+                        policy_paths = get_policy_paths(
+                            replication.policy.name,
+                            replication.policy.instance.name,
+                        )
```

Comment:
Hmmm, I don't really understand why you're copying the policy paths too... My understanding is that we were only going to copy the jenkins secrets... 🤔